ablySpec/RealtimeClientChannel.swift

```diff
@@ -550,6 +550,64 @@ class RealtimeClientChannel: QuickSpec {
 
             }
 
+            // RTL7
+            context("subscribe") {
+
+                // RTL7d
+                pending("should deliver the message even if there is an error while decoding") {
+
+                    for cryptoTest in [CryptoTest.aes128, CryptoTest.aes256] {
+                        it("using \(cryptoTest) ") {
+                            let options = AblyTests.commonAppSetup()
+                            options.autoConnect = false
+                            let client = ARTRealtime(options: options)
+                            client.setTransportClass(TestProxyTransport.self)
+                            client.connect()
+                            defer { client.close() }
+
+                            let (keyData, ivData, messages) = AblyTests.loadCryptoTestData(cryptoTest)
+                            let testMessage = messages[0]
+
+                            let cipherParams = ARTCrypto.defaultParamsWithKey(keyData, iv: ivData)
+                            let channelOptions = ARTChannelOptions(encrypted: cipherParams)
+                            let channel = client.channels.get("test", options: channelOptions)
+
+                            let transport = client.transport as! TestProxyTransport
+
+                            transport.beforeProcessingSentMessage = { protocolMessage in
+                                if protocolMessage.action == .Message {
+                                    expect(protocolMessage.messages![0].data as? String).to(equal(testMessage.encrypted.data))
+                                    expect(protocolMessage.messages![0].encoding).to(equal(testMessage.encrypted.encoding))
+                                }
+                            }
+
+                            transport.beforeProcessingReceivedMessage = { protocolMessage in
+                                if protocolMessage.action == .Message {
+                                    expect(protocolMessage.messages![0].data as? String).to(equal(testMessage.encrypted.data))
+                                    expect(protocolMessage.messages![0].encoding).to(equal(testMessage.encrypted.encoding))
+                                    // Force an error decoding a message
+                                    protocolMessage.messages![0].encoding = "bad_encoding_type"
+                                }
+                            }
+
+                            waitUntil(timeout: testTimeout) { done in
+                                channel.subscribeToName(testMessage.encoded.name) { message, errorInfo in
+                                    expect(errorInfo).toNot(beNil())
```

Comment:
I don't believe errorInfo should exist in the API anyway for a message listener, instead `an error message will be sent to the logger` and `the message will still be delivered with last successful decoding and the encoding field`. See http://docs.ably.io/client-lib-development-guide/features/#RSL6b.  I believe this test is wrong and should be checking that the encoding field is still intact i.e. it could not be decoded, and that an error is logged to the logger (if you can do this)